server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivation.java

```diff
@@ -0,0 +1,234 @@
+            assertEquals("v1", cache.get("k1"));
+            assertEquals("v2", cache.get("k2"));
+            //when the entry is removed from the cache, it should be also removed from the cache store (the store
+            //and the cache are the same sets of keys)
+            cache.remove("k1");
+            assertNull(table.getValueByKey("k1"));
+            cache.clear();
+        }
+
+    }
+
+    @Test(timeout = 600000)
+    public void testPreloadWithoutPurgePassivationFalse() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC, database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        try(TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder())) {
+            cache.clear();
+            cache.put("k1", "v1");
+            cache.put("k2", "v2");
+
+            assertNotNull(table.getValueByKey("k1"));
+            assertNotNull(table.getValueByKey("k2"));
+
+            SERVERS.getServerDriver().stop(0);
+            SERVERS.getServerDriver().restart(0);
+
+            // test preload==true, entries should be immediately in the cache after restart
+            assertEquals(2, cache.size());
+            // test purge==false, entries should remain in the database after restart
+            assertNotNull(table.getValueByKey("k1"));
+            assertNotNull(table.getValueByKey("k2"));
+        }
+;
+    }
+
+    /*
+     * This should verify that DefaultTwoWayKey2StringMapper on server side can work with ByteArrayKey which
+     * is always produced by HotRod client regardless of type of key being stored in a cache.
+     */
+    @Test(timeout = 600000)
+    public void testStoreDataWithHotRodClientPassivationFalse() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC, database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<Object, Object> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        try(TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder())) {
+            Double doubleKey = 10.0;
+            Double doubleValue = 20.0;
+
+            assertEquals(0, cache.size());
+            assertTrue(table.countAllRows() == 0);
+            cache.put(doubleKey, doubleValue);
+            // test passivation==false, database should contain all entries which are in the cache
+            assertEquals(1, table.countAllRows());
+            assertEquals(doubleValue, cache.get(doubleKey));
+        }
+    }
+
+    @Test(timeout = 600000)
+    public void testPassivateAfterEvictionPassivationEnabled() throws Exception {
```

Comment:
This test is identical to the contents of `testSoftRestartWithoutPreloadPassivationEnabled` AFAICT, except it does not have the restart. `testSoftRestartWithoutPreloadPassivationEnabled` should be sufficient`